src/pages/CaretakerProfile.tsx

```diff
@@ -0,0 +1,333 @@
+import React, { useEffect, useState } from "react";
+import store from "../store/RootStore";
+import "../scss/pages/_profile.scss";
+import testImg from "../../public/pet_buddy_logo.svg";
+import { Button, Card, Rate } from "antd";
+import CommentContainer from "../components/CommentContainer";
+import RoundedLine from "../components/RoundedLine";
+import { useLocation, useNavigate } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { api } from "../api/api";
+import { CaretakerDetails, CaretakerRatingsResponse } from "../types";
+import OfferCard from "../components/Offer/OfferCard";
+
+const CaretakerProfile: React.FC = () => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { userEmail } = location.state || {};
+  const [profileData, setProfileData] = useState<CaretakerDetails>();
+
+  const [isMyProfile, setIsMyProfile] = useState<boolean | null>(null);
+
+  const [page, setPage] = useState<number>(0);
+  const size = 10;
+  const [ratings, setRatings] = useState<CaretakerRatingsResponse>();
+
+  const getCaretakerDetails = (email: string) => {
+    api.getCaretakerDetails(email).then((data) => {
+      setProfileData(data);
+    });
+  };
+
+  const getCaretakerRatings = (
+    email: string,
+    page: number | null,
+    size: number | null
+  ) => {
+    api.getCaretakerRatings(email, page, size, null, null).then((data) => {
+      setRatings({ ...data });
+    });
+  };
+
+  useEffect(() => {
+    if (isMyProfile !== null) {
+      getCaretakerRatings(
+        isMyProfile === true ? store.user.profile!.selected_profile : userEmail,
+        page,
+        size
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [page]);
+
+  useEffect(() => {
+    store.selectedMenuOption = "profile";
+
+    //if user is visiting their profile
+    if (
+      userEmail === null ||
+      userEmail === undefined ||
+      userEmail === store.user.profile?.selected_profile
+    ) {
```

Comment:
Dlaczego jeśli email jest `null` lub `undefined` to uznajemy, że to profil aktualnego użytkownika? Chyba powinien zostać sam 3 warunek: 
```suggestion
    if (userEmail === store.user.profile?.selected_profile) {
```
Chyba, że w momencie jak przechodzimy do swojego profilu to niekoniecznie musimy przekazywać userEmail w location, ale i tak nie jestem pewny czy aż 3 warunki są tu potrzebne.

Reply:
Tak, chciałem po prostu nie musieć podawać userEmail, gdy przekierowuję na nasz profil, ale faktycznie wystarczy ten 3. warunek, więc naniosę zmianę